src/app/modules/color-scheme/color-bar/color-bar.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { ColorScheme } from '../color-schemes';
+
+/**
+ * Color bar component for the color scheme popup and color scheme selector
+ */
+@Component({
+  selector: 'ccf-color-bar',
+  templateUrl: './color-bar.component.html',
+  styleUrls: ['./color-bar.component.scss']
+})
+export class ColorBarComponent implements OnInit {
+
+  /**
+   * Color scheme to be displayed in the color bar
+   */
+  @Input() colorScheme: ColorScheme;
+
+  @Input() selected = false;
```

Comment:
How do you tell the component to select a specific color for discrete schemes?
Maybe allow this to be an index into the scheme's color array? Have to consider how it works with gradients in that case.

Reply:
I think that's how it works already? colorSchemeIndex gets updated when a color is picked and is used to find the corresponding color in the color array.

As for gradients, we could treat them like discrete schemes and give them the same kind of color arrays. Just need to look up the appropriate color values.

Reply:
How do you set an initial selection? Say I want to have the 3rd color highlighted when the component is created.

Reply:
I don't think that's been set up yet. Right now it only defaults to the first color in the array. I can add in another input for it though.